hub/api/dataset.py

```diff
@@ -101,6 +110,20 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def flush(self):
+        """Necessary operation after writes if a cache is being used. Clears the cache and sends the data to underlying storage"""
+        self.storage.flush()
+
+    def clear_cache(self):
```

Comment:
i don't understand why we need a separate function for this? shouldn't this logic be moved into the `ds.flush()` function? or better yet, the `storage.flush()` function

Reply:
we should be very careful with the functions we place in the API, ideally all methods in here should be reasonably called by the user (not just internally)

Reply:
Flushing doesn't delete data from the cache. If you're using a storage cache, stuff will continue to pile up if you use multiple datasets. I've elaborated a couple of use cases in the docstring.

Reply:
The reason why flush doesn't delete data from cache is that keeping data in the cache is good for fast retrieval.